plugins/modules/ome_alert_policies.py

```diff
@@ -0,0 +1,940 @@
+    if module.params.get('category'):
+        inp_catalog_list = module.params.get('category')
+        cdict_ref = get_category_data_tree(rest_obj)
+        if not cdict_ref:
+            module.exit_json(failed=True, msg="Failed to fetch Category details.")
+        payload_cat_list = []
+        for inp_catalog in inp_catalog_list:
+            new_dict = {}
+            catalog_name = inp_catalog.get('catalog_name')
+            if catalog_name in cdict_ref:
+                new_dict["CatalogName"] = catalog_name
+                payload_cat = []
+                category_det = cdict_ref.get(catalog_name)
+                key_id = list(category_det.keys())[0]
+                payload_subcat = []
+                category_list = inp_catalog.get('catalog_category')
+                if category_list:
+                    for inp_category in category_list:
+                        if inp_category.get('category_name') in category_det:
+                            resp_category_dict = category_det.get(
+                                inp_category.get('category_name'))
+                            key_id = list(resp_category_dict.keys())[0]
+                            sub_cat_dict = resp_category_dict.get(key_id)
+                            inp_sub_cat_list = inp_category.get('sub_category_names')
+                            if inp_sub_cat_list:
+                                for sub_cat in inp_sub_cat_list:
+                                    if sub_cat in sub_cat_dict:
+                                        payload_cat.append(key_id)
+                                        payload_subcat.append(
+                                            sub_cat_dict.get(sub_cat))
+                                    else:
+                                        module.exit_json(
+                                            failed=True,
+                                            msg=f"Sub category '{sub_cat}' in category '{inp_category.get('category_name')}' does not exist.")
+                            else:
+                                payload_cat.append(key_id)
+                                payload_subcat.append(0)
+                        else:
+                            module.exit_json(
+                                failed=True,
+                                msg=f"Category '{inp_category.get('category_name')}' in catalog '{catalog_name}' does not exist.")
+                else:
+                    payload_cat.append(0)
+                    payload_subcat.append(0)
+                new_dict["Categories"] = payload_cat
+                new_dict['SubCategories'] = payload_subcat
+            else:
+                module.exit_json(failed=True, msg=f"Catalog '{catalog_name}' does not exist.")
+            payload_cat_list.append(new_dict)
+        cat_payload['Catalogs'] = payload_cat_list
+    else:
+        mlist = []
+        if module.params.get('message_file'):
+            csvpath = module.params.get('message_file')
```

Comment:
use walrus operator if possible in line 766, 767.

Reply:
that is released 3.8 python onwards, so it will fail for <3.8

Reply:
but we are supporting minimum python version 3.9.6, isn't it ? so it's expected user has this python version as minimum requirements.

Reply:
but this is better for backward compatibility in case users are on <3.8 should not crash, no much performance difference.